src/components/application_manager/test/commands/mobile/system_request_test.cc

```diff
@@ -128,6 +144,135 @@ TEST_F(SystemRequestTest, Run_HTTP_FileName_no_binary_data_REJECTED) {
+
+  smart_objects::SmartObjectSPtr result;
+  EXPECT_CALL(app_mngr_, ManageHMICommand(_))
+      .WillOnce(DoAll(SaveArg<0>(&result), Return(true)));
+
+  SharedPtr<SystemRequest> command(CreateCommand<SystemRequest>(msg));
+  ASSERT_TRUE(command->Init());
+  command->Run();
+
+  EXPECT_EQ(MessageType::kRequest,
+            (*result)[am::strings::params][am::strings::message_type].asInt());
+  EXPECT_EQ(
+      mobile_apis::RequestType::OEM_SPECIFIC,
+      (*result)[am::strings::msg_params][am::strings::request_type].asInt());
+  EXPECT_EQ(
+      request_subtype,
+      (*msg)[am::strings::msg_params][am::strings::request_subtype].asString());
+}
+
+#ifdef PROPRIETARY_MODE
+TEST_F(
+    SystemRequestTest,
+    Run_RequestTypeAllowedAndRequestSubTypeDisallowed_SendDisallowedResponse) {
+  PreConditions();
+  MockPolicySettings mock_policy_settings;
+  PolicyHandler policy_handler(mock_policy_settings, app_mngr_);
+  utils::SharedPtr<MockPolicyManager> mock_policy_manager_ptr =
+      utils::MakeShared<MockPolicyManager>();
+  policy_handler.SetPolicyManager(mock_policy_manager_ptr);
+
+  EXPECT_CALL(app_mngr_, GetPolicyHandler())
+      .WillOnce(ReturnRef(policy_handler));
+
+  // Request type allowed
+  EXPECT_CALL(*mock_policy_manager_ptr, GetAppRequestTypesState(kAppPolicyId))
+      .WillOnce(Return(policy::RequestType::State::AVAILABLE));
+
+  const std::vector<std::string> request_types = {"OEM_SPECIFIC"};
+  EXPECT_CALL(*mock_policy_manager_ptr, GetAppRequestTypes(kAppPolicyId))
+      .WillOnce(Return(request_types));
+
+  // Request subType Allowed
+  EXPECT_CALL(*mock_policy_manager_ptr,
+              GetAppRequestSubTypesState(kAppPolicyId))
+      .WillOnce(Return(policy::RequestSubType::State::AVAILABLE));
+  const std::vector<std::string> request_subtypes = {"fakeSubType"};
+  EXPECT_CALL(*mock_policy_manager_ptr, GetAppRequestSubTypes(kAppPolicyId))
+      .WillOnce(Return(request_subtypes));
+
+  MessageSharedPtr msg = CreateIVSUMessage();
+
+  (*msg)[am::strings::msg_params][am::strings::request_type] =
+      mobile_apis::RequestType::OEM_SPECIFIC;
+
+  // Current request subType disallowed
+  const std::string request_subtype = "fakeSubType2";
+  (*msg)[am::strings::msg_params][am::strings::request_subtype] =
+      request_subtype;
+
+  ExpectManageMobileCommandWithResultCode(mobile_apis::Result::DISALLOWED);
+  EXPECT_CALL(app_mngr_, ManageHMICommand(_)).Times(0);
+
+  SharedPtr<SystemRequest> command(CreateCommand<SystemRequest>(msg));
+  ASSERT_TRUE(command->Init());
+  command->Run();
+}
+
+TEST_F(SystemRequestTest, Run_RequestTypeDisallowed_SendDisallowedResponse) {
+  PreConditions();
+  MockPolicySettings mock_policy_settings;
+  PolicyHandler policy_handler(mock_policy_settings, app_mngr_);
+  utils::SharedPtr<MockPolicyManager> mock_policy_manager_ptr =
+      utils::MakeShared<MockPolicyManager>();
+  policy_handler.SetPolicyManager(mock_policy_manager_ptr);
+
+  // Request type Allowed
```

Comment:
@BSolonenko This comment is related to...????

Reply:
@AByzhynar Fixed in https://github.com/smartdevicelink/sdl_core/pull/2151/commits/545b189a241d4db7a71d51ce3c704b2dada5f28d